tools/istio-iptables/pkg/builder/iptables-builder-impl.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package builder
+
+import (
+	"fmt"
+	"strings"
+
+	"istio.io/istio/tools/istio-iptables/pkg/dependencies"
+)
+
+// Rule represents iptables rule - chain, table and options
+type Rule struct {
+	chain  string
+	table  string
+	params []string
+}
+
+// Rules represents iptables for V4 and V6
+type Rules struct {
+	rulesv4 []*Rule
+	rulesv6 []*Rule
+}
+
+// IptablesBuilderImpl is an implementation for IptablesBuilder interface
+type IptablesBuilderImpl struct {
```

Comment:
Please add unit tests for all this type's PR, in a subsequent PR.

Reply:
Yeah.. have it in forked repo.. will submit it next once this is merged: https://github.com/abhide/istio/pull/1